### PR TITLE
feat(typoscript) Added parser and queries for TypoScript

### PR DESCRIPTION
**Pull request by Teddytrombone**
### Features
- [x] Highlighting
- [x] Language injection (comment)
- [x] Folds
- [x] Indents
- [ ] Locals (there is no concept like locals in TypoScript)

### CODE REVIEW

File: queries/typoscript/highlights.scm [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  (condition) 
+  (condition_end) 
+  (condition_else)
+] @conditional
+
+
+[
+  "@import"
+  "INCLUDE_TYPOSCRIPT"
+] @keyword
```
Comment (lucario387) [inline queries/typoscript/highlights.scm:18]: Repeated and overridden by the capture a few lines below

File: lua/nvim-treesitter/parsers.lua [hunk 2]
```diff
@@ -1874,6 +1874,14 @@ list.typescript = {
   maintainers = { "@steelsojka" },
 }
 
+list.typoscript = {
+  install_info = {
+    url = "https://github.com/Teddytrombone/tree-sitter-typoscript", -- local path or git repo
+    files = { "src/parser.c" }, -- note that some parsers also require src/scanner.c or src/scanner.cc
```
Comment (amaanq) [inline lua/nvim-treesitter/parsers.lua:1880]:
```suggestion
    url = "https://github.com/Teddytrombone/tree-sitter-typoscript",
    files = { "src/parser.c" },
```

File: queries/typoscript/folds.scm [hunk 3]
```diff
@@ -0,0 +1,3 @@
+[
+ (block)
+] @fold
```
Comment (amaanq) [inline queries/typoscript/folds.scm:3]:
```suggestion
(block) @fold
```

File: queries/typoscript/highlights.scm [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  ">"
+  "<"
+  ":="
+  "=<"
+  (condition_bool)
+] @operator
+
+[
+  ","
+] @punctuation.delimiter
```
Comment (amaanq) [inline queries/typoscript/highlights.scm:42]:
```suggestion
"," @punctuation.delimiter
```

File: queries/typoscript/indents.scm [hunk 5]
```diff
@@ -0,0 +1,7 @@
+[
+  (block)
+] @indent.begin
```
Comment (amaanq) [inline queries/typoscript/indents.scm:3]:
```suggestion
(block) @indent.begin
```

File: queries/typoscript/indents.scm [hunk 6]
```diff
@@ -0,0 +1,7 @@
...
+
+("}") @indent.branch @indent.end
```
Comment (amaanq) [inline queries/typoscript/indents.scm:5]:
```suggestion
"}" @indent.branch @indent.end 
```

**Review by amaanq — CHANGES_REQUESTED** [review]

File: queries/typoscript/highlights.scm [hunk 7]
```diff
@@ -0,0 +1,48 @@
+(identifier) @field
```
Comment (amaanq) [inline queries/typoscript/highlights.scm:1]: are all identifiers a field? seems a bit weird but that could be the case after all

Comment (Teddytrombone) [reply queries/typoscript/highlights.scm:1]: Yes imho every identifier is a field. A TypoScript configuration is a hierarchical collection of configuration blocks (allthough "configuration" is maybe not the correct word). Each block has an identifier and contains fields. Each of this fields can also be a block of configuration items. The identifier can be the identifier of the block itself or a path in the configuration tree. So for example

```
foo {
    bar {
        test = 1
    }
}
```
is the same as `foo.bar.test = 1`
So in my opinion `foo` and `bar` and `test` are fields inside the configuration. Maybe you find `@property` a better match?

**Comment by clason** [comment]
"locals" are for plugins like nvim-treesitter-refactor. It's OK to leave them out.

File: queries/typoscript/highlights.scm (hunk 7 above)
Comment (amaanq) [reply queries/typoscript/highlights.scm:1]: thanks! field is good as is

**Review by amaanq — APPROVED** [review]

File: queries/typoscript/highlights.scm [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+[
+  "@import"
+  "INCLUDE_TYPOSCRIPT"
+] @include
+
+[
+  (comment)
+  (single_line_comment)
+] @comment
```
Comment (ObserverOfTime) [inline queries/typoscript/highlights.scm:24]:
```suggestion
] @comment @spell
```

**Review by ObserverOfTime — APPROVED** [review]

**Pull request merged**